functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,276 @@ export const processNewAllotment = functions.database
+            if (currentFile.file_name === AudioFileName) 
+                currentFile.chunks.push(chunkToStore);
+            
+            // NEW FILE --> Save old file AND add the newly read chunk to `currentFile` object            
+            else {
+                if (currentFile.file_name != null) {
+
+                    // if file object exists, add the chunks to it
+                    if (chunks[currentFile.file_name]) 
+                        chunks[currentFile.file_name].push(currentFile.chunks)
+
+                    // if this is the first occurence of the file, set its value to the list of chunks 
+                    else    
+                        chunks[currentFile.file_name] = currentFile.chunks;
+                }
+
+                currentFile.file_name = AudioFileName;
+                currentFile.chunks = [chunkToStore];
+            }                    
+        }
+        
+
+        ////////////////////////////////
+        // Processing Sheet Audio Files
+        ////////////////////////////////
+
+        Object.keys(chunks).forEach(async fileName => {
+            let currentFilechunks = chunks[fileName];
+
+            // Sorting by `beginning` time
+            currentFilechunks = await lodash.orderBy(currentFilechunks, ['beginning'], ['asc']);
+            // console.log(currentFilechunks);
+
+            let chunksToImport = [];
+            ////////////////////////////////////////////
+            // [CHECK #5] Chunks of a single audio file 
+            // should not overlap each other in timing.
+            ////////////////////////////////////////////
+            let lastEndingTime = null;
+            currentFilechunks.forEach((chunk, i) => {
+
+
+                // `Beginning` value should be less than `Ending`.
+                if (chunk.beginning > chunk.ending) {
+                    console.log(`Audio file name: ${fileName} -- Error: "Beginning" value isn't less than "Ending".`);
+                    summary.overlappingChunks.push({
+                        audioFileName: fileName,
+                        beginning: chunk.beginning,
+                        ending: chunk.ending,
+                        errorMessage: `"Beginning" value isn't less than "Ending".`
+                    });
+                }
+
+                // `beginning` of the CURRENT chunk should be GREATER than the `ending` of the LAST chunk
+                if (i !== 0 && chunk.beginning && lastEndingTime && chunk.beginning < lastEndingTime) {
+                    console.log(`Audio file name: ${fileName} -- Error: Overlapping chunks.`);
+                    summary.overlappingChunks.push({
+                        audioFileName: fileName,
+                        errorMessage: `Overlapping chunks.`
+                    });
+                }
+
+
+                // setting the value of `lastEndingTime` to be used in the next iteration
+                if (chunk.ending)
```

Comment:
All the chunks should have ending time, it must be validated earlier, isn't it? So no need to check here.

Reply:
It's validated for **importing** .. not for **processing** ... or did I get this one wrong ?

Reply:
There is only one “validation”.
So the pipeline is as follows:

1. Is the row finalized (have both resolutions)? Yes — proceed with it. No — skip silently.
1. Is the row valid? All the per-row checks should be applied here. Valid — proceed. No — issue a warning.
1. Are all rows of the same file finalized and valid? Yes — import all the rows of this file with “OK” resolution. No — don't import any row of this file.

Now should be more clear.